docs/product/guidelines/theming-new.html

```diff
@@ -0,0 +1,537 @@
+</section>
+
+<section class="stacks-section">
+    {% header "h2", "Theming API" %}
+    <p class="stacks-copy">Stacks provides an API to apply custom themes.</p>
+
+    {% header "h3", "Programmatic theme generation" %}
+    <p class="stacks-copy">Use <code class="stacks-code">.create-custom-theme-hsl-rgb-variables(@color, @tier, @modeCustom)</code> to create a custom theme.</p>
+        
+    <p class="stacks-copy">This function generates two sets of CSS variables: 1) independent h/s/l and r/g/b color variables and 2) variables at each colors stop that reference the h/s/l and r/g/b variables. Provide this function the arguments defined below to generate theme colors which will apply across Stacks.</p>
+
+    <div class="overflow-x-auto mb32" tabindex="0">
+        <table class="wmn4 s-table s-table__bx-simple">
+            <thead>
+                <tr>
+                    <th scope="col">Argument</th>
+                    <th scope="col">Type</th>
+                    <th scope="col">Default</th>
+                    <th scope="col">Description</th>
+                </tr>
+            </thead>
+            <tbody class="fs-caption">
+                <tr>
+                    <td><code class="stacks-code ws-nowrap">@color</code></td>
+                    <td>HSL, hex, or other color value</td>
+                    <td></td>
+                    <td>Color to use to generate theme values.</td>
+                </tr>
+                <tr>
+                    <td><code class="stacks-code ws-nowrap">@tier</code></td>
+                    <td>
+                        <code class="stacks-code ws-nowrap">primary</code> |
+                        <code class="stacks-code ws-nowrap">secondary</code>
+                    </td>
+                    <td><code class="stacks-code ws-nowrap">primary</code></td>
+                    <td>Color tier to generate.</td>
+                </tr>
+                <tr>
+                    <td><code class="stacks-code ws-nowrap">@modeCustom</code></td>
+                    <td>
+                        <code class="stacks-code ws-nowrap">base</code> |
+                        <code class="stacks-code ws-nowrap">dark</code>
+                    </td>
+                    <td><code class="stacks-code ws-nowrap">base</code></td>
+                    <td>The color mode the theme applies to.</td>
+                </tr>
+            </tbody>
+        </table>
+    </div>
+
+    <div class="stacks-preview _preview-none mb24">
+{% highlight css %}
+.theme-custom.themed {
+    .create-custom-theme-hsl-rgb-variables(hsl(172, 37%, 48%), primary);
+    .create-custom-theme-hsl-rgb-variables(hsl(259, 29%, 55%), secondary);
+    .create-custom-theme-hsl-rgb-variables(hsl(172, 37%, 48%), primary, dark);
```

Comment:
> Is this something that makes sense in a real world scenario?

Not really!

> If not, we should change the values to don't confuse the consumers.

On it!

Reply:
Well actually… this could make sense. It's what I'm using for the theming in the docs. Because the dark color generation kinda just flips the order of light to dark, it does work just fine.

In any case, I think consumers will _mostly_ supply different colors (or at least shades) to the `base` vs `dark` modes, so I'll make them distinct here.